.github/workflows/gradle.yml

```diff
@@ -9,70 +9,87 @@ on:
       - master
       - '[2-9]+.[0-9]+.x'
 jobs:
+
   build:
     runs-on: ubuntu-latest
+
     strategy:
+      fail-fast: false
       matrix:
-        java: ['11', '14']
+        java: [11, 17]
+
     env:
       WORKSPACE: ${{ github.workspace }}
       GRADLE_OPTS: -Xmx1500m -Dfile.encoding=UTF-8
+
     steps:
       - uses: actions/checkout@v4
-      - name: Set up JDK
-        uses: actions/setup-java@v4
+      - uses: gradle/wrapper-validation-action@v2
+      - uses: actions/setup-java@v4
         with:
-          distribution: 'adopt'
+          distribution: temurin
```

Comment:
Why are using this specific distribution?

Reply:
From: https://github.com/actions/setup-java
NOTE: AdoptOpenJDK got moved to Eclipse Temurin and won't be updated anymore. It is highly recommended to migrate workflows from adopt and adopt-openj9, to temurin and semeru respectively, to keep receiving software and security updates. See more details in the [Good-bye AdoptOpenJDK post](https://blog.adoptopenjdk.net/2021/08/goodbye-adoptopenjdk-hello-adoptium/).